staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -909,6 +910,30 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		}
 	}
 
+	if len(schema.XValidations) > 0 {
+		if !opts.requireStructuralSchema {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("x-kubernetes-validations"), schema.XValidations, "x-kubernetes-validations is only supported in structural schema"))
+		} else {
+			structural, err := structuralschema.NewStructural(schema)
+			if err == nil {
```

Comment:
if the schema for a property defines a default, does that default get checked against cel expressions defined on the property? I didn't see anything explicitly dealing with this but wasn't sure if we got it for free

Reply:
It got missed. Good catch.

I've added it along with a test in a separate commit.